src/openforms/appointments/api/tests/test_endpoints_with_jcc.py

```diff
@@ -0,0 +1,211 @@
+
+        config = JccConfig.get_solo()
+        wsdl = os.path.abspath(
+            os.path.join(
+                settings.DJANGO_PROJECT_DIR,
+                "appointments/contrib/jcc/tests/mock/GenericGuidanceSystem2.wsdl",
+            )
+        )
+        config.service = SoapServiceFactory.create(url=wsdl)
+        config.save()
+
+    def setUp(self):
+        super().setUp()
+        self._add_submission_to_session(self.submission)
+
+    @requests_mock.Mocker()
+    def test_get_locations_returns_all_locations_for_a_product(self, m):
+        m.post(
+            "http://example.com/soap11",
+            text=mock_response("getGovLocationsForProductResponse.xml"),
+        )
+
+        response = self.client.get(f"{self.endpoint}?product_id=79")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()), 1)
+        location = response.json()[0]
+        self.assertEqual(location["identifier"], "1")
+        self.assertEqual(location["name"], "Maykin Media")
+
+    def test_get_locations_returns_400_when_no_product_id_is_given(self):
+        response = self.client.get(self.endpoint)
+
+        self.assertEqual(response.status_code, 400)
+
+
+class DatesListTests(SubmissionsMixin, TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.submission = SubmissionFactory.create()
+        cls.endpoint = reverse("api:appointments-dates-list")
+
+        appointments_config = AppointmentsConfig.get_solo()
+        appointments_config.config_path = (
+            "openforms.appointments.contrib.jcc.models.JccConfig"
+        )
+        appointments_config.save()
+
+        config = JccConfig.get_solo()
+        wsdl = os.path.abspath(
+            os.path.join(
+                settings.DJANGO_PROJECT_DIR,
+                "appointments/contrib/jcc/tests/mock/GenericGuidanceSystem2.wsdl",
+            )
+        )
+        config.service = SoapServiceFactory.create(url=wsdl)
+        config.save()
+
+    def setUp(self):
+        super().setUp()
+        self._add_submission_to_session(self.submission)
+
+    @requests_mock.Mocker()
+    def test_get_dates_returns_all_dates_for_a_give_location_and_product(self, m):
+        m.post(
+            "http://example.com/soap11",
+            [
+                {"text": mock_response("getGovLatestPlanDateResponse.xml")},
+                {"text": mock_response("getGovAvailableDaysResponse.xml")},
+            ],
+        )
+
+        response = self.client.get(f"{self.endpoint}?product_id=1&location_id=1")
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.json(),
+            ["2021-08-19", "2021-08-20", "2021-08-23"],
+        )
+
+    def test_get_dates_returns_400_when_missing_query_params(self):
+        for query_param in ["", "?product_id=79", "?location_id=1"]:
+            with self.subTest(query_param=query_param):
+                response = self.client.get(self.endpoint)
```

Comment:
For the future, please use the dict-structure for query params in tests:

```py
self.client.get(self.endpoint, {"product_id": 79})
```

and the test client will format the querystring itself.